lib/datadog/tracing/sampling/span/sampler.rb

```diff
@@ -50,9 +50,31 @@ def initialize(rules = [])
           def sample!(trace_op, span_op)
             return if trace_op.sampled? && trace_op.priority_sampled?
 
-            # Return as soon as one rule matches
-            @rules.any? do |rule|
-              rule.sample!(span_op) != :not_matched
+            # Applies the first matching rule
+            @rules.each do |rule|
+              decision = rule.sample!(span_op)
+
+              next if decision == :not_matched # Iterate until we find a matching decision
+
+              if decision == :kept
+                if trace_op.sampled?
```

Comment:
Move to trace formatter by always setting the decision maker tag in the root span of all trace segments.

Reply:
This is done!